src/request/url.rs

```diff
@@ -152,9 +152,12 @@ impl fmt::Display for Url {
         // Write the host.
         try!(self.host.fmt(formatter));
 
-        // Write the port.
-        try!(":".fmt(formatter));
-        try!(self.port.fmt(formatter));
+        // Write the port if they are not the default ports of 80 for HTTP and 443 for HTTPS.
+        if !((self.scheme == "http".to_owned() && self.port == 80) ||
+            (self.scheme == "https".to_owned() && self.port == 443)) {
```

Comment:
Wouldn't it be possible to do &self.scheme == "http" or similarly to prevent allocations?